src/providers/providerProvider.ts

```diff
@@ -0,0 +1,126 @@
+import * as vscode from 'vscode';
+import { Utils } from 'vscode-uri';
+import { ExecuteCommandParams, ExecuteCommandRequest } from 'vscode-languageclient';
+
+import { ClientHandler } from '../clientHandler';
+import { getActiveTextEditor } from '../vscodeUtils';
+
+interface ModuleProvidersResponse {
+  v: number;
+  provider_requirements: {
+    [provider: string]: {
+      display_name: string;
+      version_constraint?: string;
+      docs_link?: string;
+    };
+  };
+  installed_providers: {
+    [provider: string]: string;
+  };
+}
+
+class ModuleProvider extends vscode.TreeItem {
+  constructor(
+    public fullName: string,
+    public displayName: string,
+    public requiredVersion: string,
+    public installedVersion: string | undefined,
+    public docsLink: string | undefined,
+  ) {
+    super(`${displayName} ${requiredVersion}`, vscode.TreeItemCollapsibleState.None);
+
+    this.description = installedVersion ?? 'n.a.';
```

Comment:
I would say that _if_ we are going to display anything in case of "not installed" providers, then it should be more obvious that the provider is not installed (from the text). I think that we could as well just not display anything at all?

Then perhaps once we add the capability to install providers via context menu we can adjust the UI accordingly.

Reply:
Should we maybe then just display the version constraints in the description (and not in the label) for now?

<img width="238" alt="CleanShot 2021-11-16 at 11 46 52@2x" src="https://user-images.githubusercontent.com/45985/141971503-fcf9483b-44f7-46bb-845f-e5d48e4d1985.png">

Reply:
I think that installed versions are still useful, it's just the `n.a.` I was commenting on. I'd perhaps prefer empty string over that.

Reply:
In the Terraform Module Calls I returned an empty string if there was no version found.

If we intend to have a command to install a provider from a menu in this view, it makes sense to me for there to be nothing displayed in the description until it's installed, and in the tooltip have the constraint. To me, that would give me the visual indication that it's not installed if there is no resolved version there, and that I can right click on it or issue a command to have it installed.

Reply:
Okay, I agree with that. I've moved the required version from the label into the tooltip and kept the description empty when the provider isn't installed.

<img width="407" alt="CleanShot 2021-11-17 at 11 20 30@2x" src="https://user-images.githubusercontent.com/45985/142182460-cbd880c7-8a6b-4dbd-a9f9-e85168329c36.png">

Reply:
I like that! 👍🏻 Looks quite clean.